python/tests/integration/arcticdb/test_arctic.py

```diff
@@ -72,6 +75,35 @@ def test_library_creation_deletion(arctic_client):
         _lib = ac["pytest_test_lib"]
 
 
+def test_uri_override(moto_s3_uri_incl_bucket):
+    def _get_s3_storage_config(lib):
+        primary_storage_name = lib._nvs.lib_cfg().lib_desc.storage_ids[0]
+        primary_any = lib._nvs.lib_cfg().storage_by_id[primary_storage_name]
+        s3_config = S3Config()
+        primary_any.config.Unpack(s3_config)
+        return s3_config
+
+    wrong_uri = "s3://otherhost:test_bucket_0?access=dog&secret=cat&port=17988"
+    altered_ac = Arctic(moto_s3_uri_incl_bucket)
+    altered_ac._library_adapter = S3LibraryAdapter(wrong_uri)
+    # At this point the library_manager is still correct, so we can write
+    # and retrieve libraries, but the library_adapter has fake credentials
+    altered_ac.create_library("override_endpoint", LibraryOptions())
+    lib = altered_ac["override_endpoint"]
+    s3_storage = _get_s3_storage_config(lib)
+    assert s3_storage.endpoint == "otherhost:17988"
+    assert s3_storage.credential_name == "dog"
+    assert s3_storage.credential_key == "cat"
```

Comment:
There's no overide here - what does this return overridden values?

Reply:
They're not override at this point, I've just reset the library manager so that they are different to the endpoint and credentials of the actual storage - this library as-is wouldn't work because it's got bogus endpoint and credentials in its storage config. This is just so that I can check that they're overridden correctly at l102

Reply:
Discussed. I read this wrong - the second run uses the correct settings and so does test overriding.